core/jvmMain/src/kotlinx/serialization/internal/Platform.kt

```diff
@@ -153,6 +153,9 @@ private fun <T : Any> Class<T>.createEnumSerializer(): KSerializer<T> {
 }
 
 private fun <T : Any> Class<T>.findObjectSerializer(): KSerializer<T>? {
+    // Special case to avoid IllegalAccessException on Java11+
+    // There are no serializable objects in the stdlib anyway.
```

Comment:
Maybe we should add a link to the [corresponding issue](https://github.com/Kotlin/kotlinx.serialization/issues/2449)?

Reply:
It'll be in the commit message, but I can add it here too